plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/internal/crc/server/ui/PullSecretValidation.java

```diff
@@ -0,0 +1,126 @@
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.osgi.util.NLS;
+import org.jboss.tools.openshift.internal.common.ui.job.UIUpdatingJob;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class PullSecretValidation {
+
+	private static final int MAX_PULLSECRET_FILE_SIZE = 20 * 1024; // 20k
+
+	private final AtomicReference<String> error = new AtomicReference<>();
+	private final AtomicReference<String> file = new AtomicReference<>();
+
+	public String validate(final String file, final Consumer<String> listener) {
+		if (!hasChanged(file)) {
+			return error.get();
+		}
+		error.set(null);
+		this.file.set(file); 
+		new PullSecretValidationJob(file, listener).schedule();
+		return error.get();
+	}
+
+	private boolean hasChanged(String file) {
+		return !Objects.equals(this.file.get(), file);
+	}
+
+	private class PullSecretValidationJob extends UIUpdatingJob {
+
+		private String pullSecretFile;
+		private Consumer<String> listener;
+		
+		private PullSecretValidationJob(String pullSecretFile, Consumer<String> listener) {
+			super("Validating pull secret...");
+			this.pullSecretFile = pullSecretFile;
+			this.listener = listener;
+		}
+
+		@Override
+		protected IStatus run(IProgressMonitor monitor) {
+			String errorMessage = validate(pullSecretFile);
+			error.set(errorMessage);
+			return Status.OK_STATUS;
+		}
+
+		@Override
+		protected IStatus updateUI(IProgressMonitor monitor) {
+			listener.accept(error.get());
+			return Status.OK_STATUS;
+		}
+
+		private String validate(String file) {
+			String msg = null;
+			if (file == null || file.isEmpty() || !(new File(file)).isFile()) {
+				msg = "Please select a valid Pull Secret file. See https://cloud.redhat.com/openshift/install/crc/installer-provisioned for instructions.";
+			} else if (!new File(file).canRead()) {
+				msg = "Pull Secret file is not readable.";
+			} else {
+				msg = validateJsob(file);
+			}
+			return msg;
+		}
+
+		private String validateJsob(String file) {
+			String msg = null;
+			try {
+				if (!isValidJson(readPullSecretFile(file))) {
+					msg = NLS.bind("Pull Secret file {0} is invalid. Make sure it is valid JSON.", file);
+				}
+			} catch (IOException | UncheckedIOException e) {
+				msg = NLS.bind("Pull Secret file {0} is invalid. Make sure it is valid JSON: {1}", file, e.getLocalizedMessage());
+			}
+			return msg;
+		}
+
+		private String readPullSecretFile(String pullSecretFile) throws IOException {
+			String content = null;
+			try (Stream<String> lines = Files.lines(Paths.get(pullSecretFile))) {
+				content = lines
+						.limit(MAX_PULLSECRET_FILE_SIZE)
+						.collect(Collectors.joining());
+			}
+			return content;
+		}
+
+		private boolean isValidJson(String json) {
+			try {
+				new JSONObject(json);
+			} catch (JSONException ex) {
+				try {
+					new JSONArray(json);
+				} catch (JSONException ex1) {
+					return false;
```

Comment:
i know it's coming from the old file, but looks weird to me to actually perform a double try catch ...  maybe refactor to extract the JSONArray validation in another method can help :)